alcs-frontend/src/app/features/admin/configuration/configuration.component.ts

```diff
@@ -3,37 +3,92 @@ import {
+import { MaintenanceBannerConfirmationDialogComponent } from './maintenance-banner-confirmation-dialog/maintenance-banner-confirmation-dialog.component';
+import { MaintenanceService } from '../../../services/maintenance/maintenance.service';
 
 @Component({
   selector: 'app-configuration',
   templateUrl: './configuration.component.html',
   styleUrls: ['./configuration.component.scss'],
 })
 export class ConfigurationComponent implements OnInit {
-  maintenanceMode = 'false';
-  isEditing = false;
+  maintenanceBanner = false;
+  maintenanceBannerMessage = '';
+  maintenanceMode = false;
 
-  constructor(private adminConfigurationService: AdminConfigurationService) {}
+  constructor(
+    private dialog: MatDialog,
+    private adminConfigurationService: AdminConfigurationService,
+    private maintenanceService: MaintenanceService,
+    protected confirmationDialogService: ConfirmationDialogService,
+  ) {}
 
   ngOnInit(): void {
     this.loadConfigs();
   }
 
-  toggleEdit() {
-    this.isEditing = !this.isEditing;
+  private async updateConfig(configName: CONFIG_VALUE, configValue: string) {
+    await this.adminConfigurationService.setConfiguration(configName, configValue);
+    this.loadConfigs();
   }
 
-  async onSave() {
-    await this.adminConfigurationService.setConfiguration(CONFIG_VALUE.PORTAL_MAINTENANCE_MODE, this.maintenanceMode);
-    this.loadConfigs();
-    this.isEditing = false;
+  private getConfigValue(
+    configs: { name: CONFIG_VALUE; value: string }[],
+    configName: CONFIG_VALUE,
+    isBoolean: boolean = false,
+  ): string | boolean {
+    const configValue = configs.find((config) => config.name === configName)?.value || '';
+    return isBoolean ? configValue === 'true' : configValue;
   }
 
   private async loadConfigs() {
     const configs = await this.adminConfigurationService.listConfigurations();
     if (configs) {
-      const maintenanceConfig = configs.find((config) => config.name === CONFIG_VALUE.PORTAL_MAINTENANCE_MODE);
-      this.maintenanceMode = maintenanceConfig!.value;
+      this.maintenanceMode = this.getConfigValue(configs, CONFIG_VALUE.PORTAL_MAINTENANCE_MODE, true) as boolean;
+      this.maintenanceBanner = this.getConfigValue(configs, CONFIG_VALUE.APP_MAINTENANCE_BANNER, true) as boolean;
+      this.maintenanceBannerMessage = this.getConfigValue(
+        configs,
+        CONFIG_VALUE.APP_MAINTENANCE_BANNER_MESSAGE,
+      ) as string;
+    }
+  }
+
+  onToggleMaintenanceBanner(event: MatSlideToggleChange) {
+    if (event.checked === true) {
+      this.dialog
+        .open(MaintenanceBannerConfirmationDialogComponent, {
+          data: {
+            message: this.maintenanceBannerMessage,
+          },
+        })
+        .beforeClosed()
+        .subscribe((didConfirm) => {
+          if (didConfirm) {
+            this.updateConfig(CONFIG_VALUE.APP_MAINTENANCE_BANNER, this.maintenanceBanner.toString());
+            this.maintenanceService.setShowBanner(true);
```

Comment:
suggestion: Reverse these calls so that if showing the banner becomes true is only does so after the message is updated.

Reply:
Updated!